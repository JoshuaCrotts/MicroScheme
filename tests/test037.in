(define (append list1 list2)
    (if (null? list1)
        list2
        (cons (car list1) (append (cdr list1) list2))
    )
)

(define (reverse l)
  (if (null? l) '()
     (append (reverse (cdr l)) (list (car l)))
  )
)

(reverse '(1 2 3 4))
(reverse '("Joshua" "Viola" "Dookie" "Doodoo"))
