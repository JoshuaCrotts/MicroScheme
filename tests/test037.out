(4 3 2 1)
(Doodoo Dookie Viola Joshua)
